frontend/document/document.lua

```diff
@@ -99,33 +101,34 @@ function Document:discardChange()
     self.is_edited = false
 end
 
--- calculate partial digest of the document
--- since only PDF documents could be modified by KOReader by appending data
--- at the end of the files when highlighting, we use a non-even sampling
--- algorithm which samples with larger weight at file head and much smaller
--- weight at file tail, thus reduces the probability that appended data may change
--- the digest value.
--- Note that if PDF file size is around 1024, 4096, 16384, 65536, 262144
--- 1048576, 4194304, 16777216, 67108864, 268435456 or 1073741824, appending data
--- by highlighting in KOReader may change the digest value.
+-- calculate partial digest of the document and store in its docsettings to avoid document saving
+-- feature to change its checksum.
```

Comment:
I thought the comment has been out-of-date. Since no matter whether a PDF file size is around 1024, 4096, ..., this function will not return a different value for a single file. I can surely add them back.